### PR TITLE
ci: separate build steps in integration tests

### PR DESCRIPTION
**Pull request by vindard**
## Description

Separate build and run steps for integration tests to help with debugging and reliability

### CODE REVIEW

File: .github/workflows/integration-test.yml [hunk 1]
```diff
@@ -35,17 +36,65 @@ jobs:
...
             case "$LABEL" in
-              dashboard|consent|pay|core|admin-panel|map|voucher)
+              core|dashboard|consent|pay|admin-panel|map|voucher)
+                BUILD_ARGS+=" //core/api:prod_build"
+                BUILD_ARGS+=" //core/notifications:notifications"
+
+                break
+                ;;
+            esac
+          done
```
Comment (dolcalmi) [inline .github/workflows/integration-test.yml:49]: if we are not using #LABEL why do we need a for

File: .github/workflows/integration-test.yml [hunk 2]
```diff
@@ -35,17 +36,65 @@ jobs:
...
+
+          # Handle root 'node_modules' dependency
+          for LABEL in "${LABELS[@]}"; do
+            case "$LABEL" in
+              dashboard|consent|pay|admin-panel|map|voucher)
+                BUILD_ARGS+=" "//:node_modules""
```
Comment (dolcalmi) [inline .github/workflows/integration-test.yml:55]: same here

File: .github/workflows/integration-test.yml [hunk 3]
```diff
@@ -35,17 +36,65 @@ jobs:
...
+                ;;
+            esac
+          done
+
+          # Handle 'consent' dependencies
+          for LABEL in "${LABELS[@]}"; do
+            case "$LABEL" in
+              dashboard|voucher)
+                BUILD_ARGS+=" //apps/consent:consent"
+                BUILD_ARGS+=" //core/api-keys:api-keys"
```
Comment (dolcalmi) [inline .github/workflows/integration-test.yml:67]: and here

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: .github/workflows/integration-test.yml (hunk 2 above)
Comment (vindard) [reply .github/workflows/integration-test.yml:55]: We only execute line Ln 55 if it's one of the `app` labels right now

**Review by dolcalmi — APPROVED** [review]

**Pull request merged**